src/simu/phantom.c

```diff
@@ -517,18 +518,98 @@ static void separate_bckgrd(int Nb, struct ellipsis_s bckgrd[Nb], int Nf, struct
 	}
 }
 
-void calc_phantom_tubes(const long dims[DIMS], complex float* out, bool kspace, const long tstrs[DIMS], const complex float* traj)
+
+void calc_phantom_tubes(const long dims[DIMS], complex float* out, bool kspace, bool random, int N, const long tstrs[DIMS], const complex float* traj)
 {
 	if (1 < dims[COEFF_DIM]) {
 
+		struct ellipsis_bs phantom_tubes_N[2 * N - 1];
+
+		if (random) {
+
+			// generate random ellipse
+			for (int i = 0; i < 2 * N - 2; i+=2) {
+
+				double sx = 0;
+				double sy = 0;
+				double px = 0;
+				double py = 0;
+
+				bool overlap = true;
+
+				while (overlap) {
```

Comment:
Doesn't this go into an endless loop when N is too high and there is not enough space?

Reply:
I would just error out after a reasonable number of trials.